src/data/basic_dataset.py

```diff
@@ -131,12 +127,18 @@ def __getitem__(self, idx):
         image_path = qa["image_path"]
         system_prompt = get_system_prompt(qa["qa_type"])
 
-        return (
-            self.message_format.format(
-                question, key_object_info, image_path, system_prompt, answer
-            ),
-            question,
-            answer,
-            qa["id"],
-            qa["qa_type"],
+        query_item = QueryItem(
+            question=question,
+            image_path=image_path,
+            qa_id=qa["id"],
+            qa_type=qa["qa_type"],
+            key_object_info=key_object_info,
```

Comment:
A note / check that this field should only be used when using the train DS would be great, as this is a bit confusing otherwise.